weave/trait/telemetry/wdm/telemetry_wdm_trait.proto

```diff
@@ -0,0 +1,94 @@
+//
+//    Copyright (c) 2016 Nest Labs, Inc.
+//    All rights reserved.
+//
+//    This document is the property of Nest. It is considered
+//    confidential and proprietary information.
+//
+//    This document may not be reproduced or transmitted in any form,
+//    in whole or in part, without the express written permission of
+//    Nest.
+//
+
+//    @file
+//      This file specifies a WDM Telemetry trait that specifies the
+//      various telemetry counters that are associated with WDM
+
+syntax = "proto3";
+
+package weave.trait.telemetry.wdm;
+
+import "wdl/wdl_options.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "WeaveInternalTelemetryWdmTrait";
+option objc_class_prefix = "SCM";
+
+//  @brief
+//    TelemetryWdmTrait
+//
+//  A telemetry trait for the WDM that specifies various counters
+//  associated with WDM updates and subscriptions.
+//
+message TelemetryWdmTrait {
+  option (wdl.message_type) = TRAIT;
+  option (wdl.trait) = {
+    stability: ALPHA,
+    id: 0x1781,
+    vendor_id: 0x0000,
+    version: 1
+  };
+  option (wdl.properties) = {
+    writable: READ_ONLY,
+  };
+
+  // Events associated with WDM updates
+  message TelemetryWdmUpdateStatsEvent {
+    option (wdl.message_type) = EVENT;
+    option (wdl.event) = {
+      id: 1
+      event_importance: EVENT_IMPORTANCE_DEBUG
+    };
+
+    // This field holds the count of successful updates since the
+    // last restart.
+    uint32 successful_updates_count = 1;
+
+    // This field holds the count of unsuccessful updates since the
+    // last restart.
+    uint32 unsuccessful_updates_count = 2;
+
+    // This field holds the count of successful subscriptions since the
+    // last restart.
+    uint32 successful_subscriptions_count = 3;
+
+    // This field holds the count of subscription attempts since the
+    // last restart.
+    uint32 subscription_attempt_count = 4;
+
+    // This field holds the timestamp of the last update failure.
+    google.protobuf.Timestamp last_update_failure_time = 5 [(wdl.param) = {
+      timestamp_constraints: { signed: true, precision: 0.001, width: 64 }
+    }];
+
+    // This field holds the timestamp of the last subscription failure.
+    google.protobuf.Timestamp last_subscription_failure_time = 6 [(wdl.param) = {
+      timestamp_constraints: { signed: true, precision: 0.001, width: 64 }
+    }];
+
+
+    // This field holds the error code of the last update failure.
+    int32 last_update_failure_reason = 7;
```

Comment:
For the last failure update, it would also make some sense to note the address of the subscriber.

Reply:
done